File: Sqrilizz-Reports/src/main/java/dev/sqrilizz/SQRILIZZREPORTS/BugReportCommand.java
package dev.sqrilizz.SQRILIZZREPORTS;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Arrays;
import java.util.List;

public class BugReportCommand implements CommandExecutor {
    
    // Доступные категории багов
    private static final List<String> BUG_CATEGORIES = Arrays.asList(
        "duplication", "dupe",      // Дюп предметов
        "crash", "server-crash",    // Краш сервера
        "exploit", "glitch",        // Эксплойты
        "performance", "lag",       // Проблемы с производительностью
        "gameplay", "mechanic",     // Игровая механика
        "world", "generation",      // Генерация мира
        "inventory", "items",       // Инвентарь/предметы
        "commands", "cmd",          // Команды
        "permissions", "perms",     // Права доступа
        "economy", "money",         // Экономика
        "other", "misc"             // Прочее
    );
    
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage(LanguageManager.getMessage("player-only"));
            return true;
        }

        Player player = (Player) sender;

        if (args.length < 2) {
            VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-usage"));
            sendCategoriesList(player);
            return true;
        }

        String category = args[0].toLowerCase();
        String description = String.join(" ", Arrays.copyOfRange(args, 1, args.length));

        if (description.isEmpty()) {
            VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-usage"));
            return true;
        }

        // Проверяем валидность категории
        if (!isValidCategory(category)) {
            VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-invalid-category")
                .replace("[CATEGORY]", category));
            sendCategoriesList(player);
            return true;
        }

        // Нормализуем категорию
        String normalizedCategory = normalizeCategory(category);

        // Проверяем кулдаун
        if (CooldownManager.hasCooldown(VersionUtils.getPlayerUUID(player))) {
            long remainingTime = CooldownManager.getRemainingTime(VersionUtils.getPlayerUUID(player));
            VersionUtils.sendMessage(player, LanguageManager.getMessage("cooldown-message")
                .replace("[COOLDOWN]", String.valueOf(remainingTime)));
            return true;
        }

        // Проверяем систему защиты от злоупотреблений для багрепортов
        if (!AntiAbuseManager.canReport(player, "BUG_REPORT")) {
            return true;
        }

        // Отправляем багрепорт с категорией
        ReportManager.addBugReport(player, normalizedCategory, description);
        
        // Регистрируем багрепорт в системе защиты от злоупотреблений
        AntiAbuseManager.recordReport(player, "BUG_REPORT");
        
        // Устанавливаем кулдаун
        CooldownManager.setCooldown(VersionUtils.getPlayerUUID(player));
        
        // Уведомляем игрока об успешной отправке
        VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-success")
            .replace("[CATEGORY]", getCategoryDisplayName(normalizedCategory))
            .replace("[DESCRIPTION]", description));
        
        return true;
    }
    
    private boolean isValidCategory(String category) {
        return BUG_CATEGORIES.contains(category.toLowerCase());
    }
    
    private String normalizeCategory(String category) {
        category = category.toLowerCase();
        // Нормализуем синонимы к основным категориям
        switch (category) {
            case "dupe": return "duplication";
            case "server-crash": return "crash";
            case "glitch": return "exploit";
            case "lag": return "performance";
            case "mechanic": return "gameplay";
            case "generation": return "world";
            case "items": return "inventory";
            case "cmd": return "commands";
            case "perms": return "permissions";
            case "money": return "economy";
            case "misc": return "other";
            default: return category;
        }
    }
    
    private String getCategoryDisplayName(String category) {
        return LanguageManager.getMessage("bugreport-category-" + category);
    }
    
    private void sendCategoriesList(Player player) {
        VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-categories-header"));
        
        // Группируем категории для отображения
        String[] mainCategories = {"duplication", "crash", "exploit", "performance", "gameplay", 
                                   "world", "inventory", "commands", "permissions", "economy", "other"};
        
        for (String cat : mainCategories) {
            VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-category-item")
                .replace("[CATEGORY]", cat)
                .replace("[DESCRIPTION]", getCategoryDisplayName(cat)));
        }
    }
}

File: Sqrilizz-Reports/src/main/kotlin/dev/sqrilizz/SQRILIZZREPORTS/BugReportCommand.kt
package dev.sqrilizz.SQRILIZZREPORTS

import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class BugReportCommand : CommandExecutor {

    companion object {
        private val BUG_CATEGORIES = listOf(
            "duplication", "dupe",
            "crash", "server-crash",
            "exploit", "glitch",
            "performance", "lag",
            "gameplay", "mechanic",
            "world", "generation",
            "inventory", "items",
            "commands", "cmd",
            "permissions", "perms",
            "economy", "money",
            "other", "misc"
        )

        private val CATEGORY_ALIASES = mapOf(
            "dupe" to "duplication",
            "server-crash" to "crash",
            "glitch" to "exploit",
            "lag" to "performance",
            "mechanic" to "gameplay",
            "generation" to "world",
            "items" to "inventory",
            "cmd" to "commands",
            "perms" to "permissions",
            "money" to "economy",
            "misc" to "other"
        )
    }

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (sender !is Player) {
            sender.sendMessage(LanguageManager.getMessage("player-only"))
            return true
        }

        if (args.size < 2) {
            VersionUtils.sendMessage(sender, LanguageManager.getMessage("bugreport-usage"))
            sendCategoriesList(sender)
            return true
        }

        val category = args[0].lowercase()
        val description = args.drop(1).joinToString(" ")

        if (description.isEmpty()) {
            VersionUtils.sendMessage(sender, LanguageManager.getMessage("bugreport-usage"))
            return true
        }

        if (category !in BUG_CATEGORIES) {
            VersionUtils.sendMessage(sender, LanguageManager.getMessage("bugreport-invalid-category")
                .replace("[CATEGORY]", category))
            sendCategoriesList(sender)
            return true
        }

        val normalizedCategory = CATEGORY_ALIASES[category] ?: category

        if (CooldownManager.hasCooldown(VersionUtils.getPlayerUUID(sender))) {
            val remainingTime = CooldownManager.getRemainingTime(VersionUtils.getPlayerUUID(sender))
            VersionUtils.sendMessage(sender, LanguageManager.getMessage("cooldown-message")
                .replace("[COOLDOWN]", remainingTime.toString()))
            return true
        }

        if (!AntiAbuseManager.canReport(sender, "BUG_REPORT")) {
            return true
        }

        ReportManager.addBugReport(sender, normalizedCategory, description)
        AntiAbuseManager.recordReport(sender, "BUG_REPORT")
        CooldownManager.setCooldown(VersionUtils.getPlayerUUID(sender))

        VersionUtils.sendMessage(sender, LanguageManager.getMessage("bugreport-success")
            .replace("[CATEGORY]", LanguageManager.getMessage("bugreport-category-$normalizedCategory"))
            .replace("[DESCRIPTION]", description))

        return true
    }

    private fun sendCategoriesList(player: Player) {
        VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-categories-header"))

        val mainCategories = arrayOf(
            "duplication", "crash", "exploit", "performance", "gameplay",
            "world", "inventory", "commands", "permissions", "economy", "other"
        )

        for (cat in mainCategories) {
            val aliases = CATEGORY_ALIASES.entries
                .filter { it.value == cat }
                .joinToString(", ") { it.key }
            val aliasText = if (aliases.isNotEmpty()) " ($aliases)" else ""
            VersionUtils.sendMessage(player, LanguageManager.getMessage("bugreport-category-entry")
                .replace("[CATEGORY]", cat)
                .replace("[ALIASES]", aliasText)
                .replace("[DESCRIPTION]", LanguageManager.getMessage("bugreport-category-$cat")))
        }
    }
}
